OpenRA.Mods.Common/ServerTraits/MasterServerPinger.cs

```diff
@@ -115,5 +138,33 @@ public void PingMasterServer(S server)
 
 			a.BeginInvoke(null, null);
 		}
+
+		void CreateLANGameBeacon(S server, int numPlayers, int numSlots, int numBots, int numSpectators, Manifest mod)
+		{
+			var settings = server.Settings;
+			var lanGameYaml =
+@"Game:
+	Id: {0}
+	Name: {1}
+	Address: 0.0.0.0:{2}
+	State: {3}
+	Players: {4}
+	MaxPlayers: {5}
+	Bots: {6}
+	Spectators: {7}
+	Map: {8}
+	Mods: {9}@{10}
+	Protected: {11}".F(currentProcessId, settings.Name, settings.ListenPort, (int)server.State, numPlayers, numSlots, numBots, numSpectators,
+				server.Map.Uid, mod.Id, mod.Metadata.Version, !string.IsNullOrEmpty(settings.Password));
+
+			if (lanGameYaml != lastLanGameBeacon)
```

Comment:
It would be a lot cleaner to check for state changes at a higher level (or not - is there a downside to recreating beacons periodically?) and then have that apply to both LAN and master server updates.